### DISCUSSION TITLE
MQTT working only for a simple button

### DISCUSSION THREAD

**Discussion by therionrg** [HASwitchPlate/openHASP#468] [Q&A] [answered]
Hello,

If I have a simple button like: 
> {"obj":"btn","id":186,"text":"click","x":180,"y":80,"w":120,"h":60}

If I click the button I can find in my MQTT Explorer the published topic hasp/plate/state/p3b186 and the 2 events for a button click. So generally MQTT is configured properly and working fine.

But now when I make a toggle button like:
> {"obj":"btn","id":186,"text":"toggle","toggle":true,"x":180,"y":80,"w":120,"h":60}

then I cannot find any published MQTT topics, like there was no publishing. The same issue I have for the switch object.

Running a serial terminal for debug purpose shows me, that the MQTT topic for the button press is published. Also if I check the published MQTT topics in the Information page the counter got increased by 2.

I hope anyone can give me any hints, because this is really a great project and I would like to do more panels, but only with buttons it's complicated.

kind regards
Roger

**Comment by fvanroie** [comment, marked as answer]
> {"obj":"btn","id":186,"text":"toggle","toggle":true,"x":180,"y":80,"w":120,"h":60}

I tested this jsonl and for me it worked just fine:
![image](https://user-images.githubusercontent.com/15969459/232115072-66227d2d-f7b4-4a7c-9e48-109acac5621a.png)

The events are `up` and `down` with a `val` for the current state.

What version and board are you running?

Also make sure that the `action` property is not set on the toggle object, as that prevents the MQTT events from being sent. Click Clear Pages and then Reload Pages in the file editor or reboot the device to make sure.

**Reply by therionrg** [reply]
Thanks a lot for your help, board is a wt32-sc01 with v0.6.3. I found out that my issuse is not related to openHASP. Because if I install a mosquitto server on my machine I can see all the published topics without any issue. 
So I guess I have to dig into the iobroker mqtt server to solve that issue....
